packages/orm/src/EntityManager.ts

```diff
@@ -986,7 +1009,9 @@ export function sameEntity(a: Entity, bMeta: EntityMetadata<any>, bCurrent: Enti
 export function getMetadata<T extends Entity>(entity: T): EntityMetadata<T>;
 export function getMetadata<T extends Entity>(type: EntityConstructor<T>): EntityMetadata<T>;
 export function getMetadata<T extends Entity>(entityOrType: T | EntityConstructor<T>): EntityMetadata<T> {
-  return (isEntity(entityOrType) ? entityOrType.__orm.metadata : (entityOrType as any).metadata) as EntityMetadata<T>;
+  return (typeof entityOrType === "function"
+    ? (entityOrType as any).metadata
+    : entityOrType.__orm.metadata) as EntityMetadata<T>;
```

Comment:
Not sure it really matters vs. the `O(N^2)` fix, but flipped it to check "is this entity constructor" first b/c it's a much simpler `typeof` check vs. the ~2-3 probes that `isEntity` does.